### PR TITLE
[#3624] - add a basic text to tell user if he/she is offline

### PR DESCRIPTION
**Pull request by marvinkome**
#### Before the PR (what is the issue or what needed to be done)
Tell user is he/she is online or offline

#### The solution
This is a simple implementation of using navigator.onLine.
It wouldn't work on some older browsers, but we would handle this edge case in future modifications

#### Screenshots (if appropriate)
image

#### Reviewer Checklist
* [ ] Added an explanation about the work done
* [ ] Connected the PR and the issue on Zenhub
* [ ] Added a test plan to the issue
* [ ] Updated the copyright header (when relevant)
* [ ] Formatted the code
* [ ] Added a documentation (if relevant)
* [ ] Added some unit tests (if relevant)

### CODE REVIEW

**Review by iperdomo — CHANGES_REQUESTED** [review]
The `onlineStatus` is a *new* API in the browsers. If you're connected to the local office WiFi you'll be *online* but potentially you can't reach the Flow Dashboard. See the comments on the other PR - https://github.com/akvo/akvo-flow/pull/3625#issuecomment-693384279

Some apps reach to an known service (e.g. Google) to know if you're actually online.

> Unfortunately, these events aren't fully reliable. If you need greater reliability, or if the API isn't implemented in the browser, you can use other signals to detect if you are offline including using service workers and responses from XMLHttpRequest.

https://developer.mozilla.org/en-US/docs/Web/API/NavigatorOnLine/Online_and_offline_events

> "online" does not always mean connection to the internet, it can also just mean connection to some network. 

https://caniuse.com/online-status

**Comment by marvinkome** [comment]
@iperdomo based on this. Do I remove the current navigator.onLine implementation and replace it with a timeout that pings google every X seconds?

**Comment by iperdomo** [comment]
> @iperdomo based on this. Do I remove the current navigator.onLine implementation and replace it with a timeout that pings google every X seconds?

Yes, I think is a *safer* approach

**Review by iperdomo — CHANGES_REQUESTED** [review]
Still waiting for the changes

File: Dashboard/app/js/lib/components/reusable/OfflineIndicator/index.jsx [hunk 1]
```diff
@@ -0,0 +1,31 @@
...
+
+  componentDidMount() {
+    this.timer = setInterval(() => {
+      fetch('https://www.google.com/generate_204', { mode: 'no-cors' })
+        .then(res => {
+          // with no-cors, res.status is 0 because the javascript ignores all response.
+          // we don't need the response, all we need to know is if the request went through or not
+          if (res.status !== 204) {
+            this.setState({ isOffline: false });
+          }
```
Comment (marvinkome) [inline Dashboard/app/js/lib/components/reusable/OfflineIndicator/index.jsx:16]: https://stackoverflow.com/questions/58518098/fetch-request-to-httpswww-google-com-fails-obviously-with-cors#comment103363447_58518361

This explains this line

File: Dashboard/app/js/lib/components/reusable/OfflineIndicator/index.jsx [hunk 2]
```diff
@@ -6,19 +6,25 @@ export default class OfflineIndicator extends React.Component {
...
 
   componentDidMount() {
-    window.addEventListener('online', this.onlineStatus);
-    window.addEventListener('offline', this.onlineStatus);
+    this.timer = setInterval(() => {
+      fetch('https://www.google.com/generate_204', { mode: 'no-cors' })
+        .then(res => {
+          // with no-cors, res.status is 0 because the javascript ignores all response.
+          // we don't need the response, all we need to know is if the request went through or not
+          if (res.status !== 204) {
```
Comment (iperdomo) [inline Dashboard/app/js/lib/components/reusable/OfflineIndicator/index.jsx:14]: We want to check if the status is 204? `res.status === 204` right?

Comment (iperdomo) [reply Dashboard/app/js/lib/components/reusable/OfflineIndicator/index.jsx:14]: OK, i tested it, then if we know that `res.status` is `0`, we should check for that? `res.status === 0` ?

Comment (iperdomo) [reply Dashboard/app/js/lib/components/reusable/OfflineIndicator/index.jsx:14]: So either check for `res.status === 0` or remove the whole `if` block, since we don't want to check the status

**Review by iperdomo — APPROVED** [review]

File: Dashboard/app/js/lib/components/reusable/OfflineIndicator/index.jsx [hunk 3]
```diff
@@ -0,0 +1,27 @@
...
+        });
+    }, 5000);
+  }
+
+  componentWillUnmount() {
+    clearInterval(this.timer);
+  }
+
+  render() {
+    return this.state.isOffline && <p>You are offline. Changes {"won't"} be saved</p>;
```
Comment (iperdomo) [inline Dashboard/app/js/lib/components/reusable/OfflineIndicator/index.jsx:25]: @marvinkome what about translations of this messages?
